bundle/terranova/apply.go

```diff
@@ -57,10 +57,17 @@ func (m *terranovaApplyMutator) Apply(ctx context.Context, b *bundle.Bundle) dia
 			return
 		}
 
-		config, ok := b.GetResourceConfig(node.Group, node.Name)
-		if !ok {
-			diags.AppendErrorf("internal error: cannot get config for %s", node)
-			return
+		var config any
+
+		if node.ActionType == deployplan.ActionTypeDelete {
+			config = nil
+		} else {
+			var ok bool
+			config, ok = b.GetResourceConfig(node.Group, node.Name)
+			if !ok {
+				diags.AppendErrorf("internal error: cannot get config for group=%v name=%v", node.Group, node.Name)
+				return
+			}
 		}
```

Comment:
Why is this necessary now and not before? (useful to capture in a comment here as well)

I suspect it's because the state loader now populates configs (and IDs) regardless of presence in the config.